src/vs/workbench/contrib/positronPlots/browser/components/webviewPlotInstance.tsx

```diff
@@ -29,12 +29,15 @@ export const WebviewPlotInstance = (props: WebviewPlotInstanceProps) => {
 	useEffect(() => {
 		const client = props.plotClient;
 		client.claim(this);
-		if (webviewRef.current) {
-			client.layoutWebviewOverElement(webviewRef.current);
-		}
 		return () => {
 			client.release(this);
 		};
+	}, [props.plotClient]);
+
+	useEffect(() => {
+		if (webviewRef.current) {
+			props.plotClient.layoutWebviewOverElement(webviewRef.current);
+		}
 	});
```

Comment:
Since `useEffect` runs on the next tick this _may_ cause stuttering. Unfortunately, the typical solution is to use `useLayoutEffect()` but that doesn't work here because the very first render cycle the `webviewRef` won't be available (and thus the plot won't show). Another solution that technically would remove a single frame of latency in resizing is to just run the layout code in the main component function while also calling it in the main `useEffect` so it works the first render. 

```tsx
...
useEffect(() => {
	const client = props.plotClient;
	client.claim(this);
	if (webviewRef.current) {
		props.plotClient.layoutWebviewOverElement(webviewRef.current);
	}
	return () => {
		client.release(this);
	};
}, [props.plotClient]);


if (webviewRef.current) {
	props.plotClient.layoutWebviewOverElement(webviewRef.current);
}
..
```

This is all a bit academic though and is probably more trouble than it's worth!

Reply:
That makes sense! I'll leave it out for now, but I do appreciate the write-up, always good to learn more React tricks.